File: roadmap/c/demo/02_variables.c
/* 02_variables.c — types and sizeof.
 * Build: gcc -std=c11 -Wall -Wextra -Werror 02_variables.c -o variables
 */
#include <stdint.h>    /* fixed-width types */
#include <stdio.h>

int main(void) {
    int       age    = 30;            /* signed 32-bit integer */
    double    height = 1.82;          /* double precision float */
    char      letter = 'A';           /* one character */
    uint32_t  count  = 65535U;        /* exactly 32 unsigned bits */
    /* size_t is the natural type for sizes - %zu prints it */
    printf("int:   %d\n", age);
    printf("double:%.2f\n", height);
    printf("char:  %c\n", letter);
    printf("uint32:%u\n", count);
    printf("sizeof(int)  = %zu bytes\n", sizeof(int));
    printf("sizeof(double)= %zu bytes\n", sizeof(double));
    return 0;
}

File: roadmap/csharp/demo/02_variables.cs
// Variables and basic data types
int age = 30;                  // integer
double price = 19.99;          // double precision float
decimal balance = 1000.50m;    // decimal — exact, for money
bool enabled = true;           // boolean
char letter = 'A';             // single character
string name = "Ada";           // text

var count = 7;                 // inferred type: int
int? maybe = null;             // nullable int

Console.WriteLine($"{name} is {age}, balance {balance}, enabled {enabled}");
Console.WriteLine($"count={count}, letter={letter}, maybe={maybe ?? -1}");
